eng/platform-matrix-combos.yml

```diff
@@ -0,0 +1,71 @@
+parameters:
+  jobTemplate: ''
+  buildConfig: ''
+  # helixQueueGroup is a named collection of Helix Queues. If specified, it determines which Helix queues are
+  # used, instead of the usual criteria. Allowed values:
+  # 'pr' - the queues used for a pull request for the platform. Typically a small set.
+  # 'ci' - the queues used for a CI (post-merge) test run.
+  # 'all' - the queues used for non-PR, non-CI test runs, e.g., Manual or Scheduled runs. Typically this is all available queues.
+  # 'corefx' - the queues used for a corefx test run.
+  helixQueueGroup: 'pr'
+  stagedBuild: false
+  jobParameters: {}
+  linuxArm: false
+  linuxArm64: false
+  linuxMuslX64: false
+  linuxMuslArm64: false
+  linuxRhel6X64: false
+  linuxX64: false
+  osxX64: false
+  windowsX64: false
+  windowsX86: false
+  windowsArm: false
+  windowsArm64: false
+
+jobs:
+- template: /eng/platform-matrix-managed-test-build.yml
+  parameters:
+    jobTemplate: ${{ parameters.jobTemplate }}
+    buildConfig: ${{ parameters.buildConfig }}
+    helixQueueGroup: ${{ parameters.helixQueueGroup }}
+    stagedBuild: ${{ parameters.stagedBuild }}
+    jobParameters: ${{ parameters.jobParameters }}
+
+    linuxArm: ${{ parameters.linuxArm }}
```

Comment:
Maintaining this list of platform/arch combos is potentially expensive for us long term. What was the complication that lead to explicitly setting variables like them, instead of passing a string?

Reply:
The reason was basically to avoid duplication. The calculated combos are used for two purposes - to check whether a particular combo should be built and to pick suitable OS'es for building the managed test artifacts. It didn't occur to me that we can reuse managed test artifacts built for a different architecture (and I suspect we cannot in R2R runs as I believe that the R2R compilation takes place during test build). Without using the temporary variables we would need to repeat the lengthy checks we use for calculating them in both places.

One other option I experimented with during the development of this change was just using the platform-matrix.yml template to basically explode platformGroups into platforms so that 'platforms' always ends up as a plain linear list of all combos to build & run. This subsequently simplifies the checks in the inner template. My thinking was that using separate variables is slightly safer as the Azure DevOps parser will bail out in case of typos rather than silently skipping a combo in case of a typo in string comparisons.

If you find the "platforms" option preferable, I can easily do that but, exactly as you said, considering all the completed testing and other infra work depending on or conflicting with my change I would prefer to merge it in as is and follow up in a separate change.